build/component/__fixture__.mjs

```diff
@@ -15,30 +24,30 @@ const writeInstanceAsync = async (
   { root, conf, main },
   config,
 ) => {
-  const directory = `${root}/${component}/${instance}`;
+  const directory = joinPath(root, component, instance);
   await mkdir(directory);
-  await writeFile(`${directory}/${conf}`, stringifyYAML(config), "utf8");
+  await writeFile(joinPath(directory, conf), stringifyYAML(config), "utf8");
```

Comment:
Seeing this reminded me of something I've been meaning to ask about: is there a reason to prefer, e.g. `await writeFile` to `writeFileSync`?

Reply:
Performing synchronous IO operations is perceived as a bad practice. Node is pushing hard to avoid those. Doing asynchronous operations alleviate the biggest weakness of having a single thread: performance. With asynchronous operation, the JS engine can delegate work to a pool of thread that is not responsible to execute JS code. But, sometimes, performing synchronous IOs *is* the right way to go. This is the case for instance when doing meta stuff like we do.